test/ParallelIOTest.cpp

```diff
@@ -85,6 +85,7 @@ void write_test_zero_extent( bool fileBased, std::string file_ending, bool write
         e["position"]["x"].storeChunk(position_local, {offset}, {rank});
         e["positionOffset"]["x"].storeChunk(positionOffset_local, {offset}, {rank});
     }
+    o.flush();
```

Comment:
If one removes this flush now, this test hangs in `ParallelADIOS1IOHandlerImpl::open_write()` (one rank) and in `~ParallelADIOS1IOHandlerImpl` in `MPI_Barrier(m_mpiComm);` (the other rank)